tests/test_get_cve.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import subprocess
+from subprocess import run
+import tempfile
+import lib.get_cve
+from lib.get_cve import git_log_titles, find_log_titles
+from pathlib import Path
+
+mock_list_titles_1 = [
+    "Add memory requirements",
+    "fix(XX-3444): test1",
+    "fix(XX-3445): test2",
+    "feat(XX-3446): test3",
+]
+
+mock_list_titles_2 = [
+    "Add memory requirements",
+    "fix(CVE-3444): test1",
+    "fix(CVE-3445): test2",
+    "feat(CVE-3446): test3",
+]
+
+mock_result_good = [
+    "fix(CVE-3444): test1",
+    "fix(CVE-3445): test2"
+]
+mock_reponse_data_empty = []
+
+
+class MockCompletedProcess:
+    def __init__(self, returncode, stdout, stderr=""):
+        self.returncode = returncode
+        self.stdout = stdout
+        self.stderr = stderr
+
+
+def mock_empty_log_titles():
+    return []
+
+
+def test_git_log_empty_list(monkeypatch):
+    monkeypatch.setattr(lib.get_cve, 'find_log_titles', mock_empty_log_titles)
+    result = find_log_titles(mock_list_titles_1)
```

Comment:
There are two things in here...

1. If you change `mock_list_titles_1` by `mock_list_titles_2` the test fails, because it finds titles. That means that if you wanted to parch the executing with the `monkeypatch.setattr` it is not working, the same principle as with `subprocess.run` applies.

2. Why are you patching `find_log_titles`? The test should be executing that function, so we can check it works properly. By patching it we are not really running it, so what is the point?

Reply:
I will remove this